tests/utils/test_rest_utils.py

```diff
@@ -602,3 +602,45 @@ def test_provide_redirect_kwarg():
             verify=mock.ANY,
             timeout=120,
         )
+
+
+def test_http_request_max_retries(monkeypatch):
+    monkeypatch.setenv("_MLFLOW_HTTP_REQUEST_MAX_RETRIES_LIMIT", "15")
+    host_creds = MlflowHostCreds("http://example.com")
+
+    with mock.patch("requests.Session.request") as mock_request:
+        with pytest.raises(MlflowException, match="The configured max_retries"):
+            http_request(host_creds, "/endpoint", "GET", max_retries=16)
+        mock_request.assert_not_called()
+        http_request(host_creds, "/endpoint", "GET", max_retries=3)
+        mock_request.assert_called()
```

Comment:
```suggestion
        mock_request.assert_called_once()
```

nit: for more strict validation.